caikit_nlp/modules/text_generation/peft_prompt_tuning.py

```diff
@@ -1060,21 +1063,27 @@ def _execute_train_loop(
                         optimizer.step()
                         lr_scheduler.step()
                         optimizer.zero_grad()
+                        step_loss_log[step_count] = loss
+                        step_count += 1
                 except torch.cuda.OutOfMemoryError:
                     error(
                         "<NLP07175292E>",
                         MemoryError("Not enough memory available for training!"),
                     )
 
             log.info("<NLP46114010I>", {"loss": float(loss), "epoch": epoch})
-            # Below is added to be propagated and stored as training_metadata
-            training_loss_tracker.append(
-                {
-                    "epoch": epoch,
-                    "value": float(loss),
-                    "timestamp": datetime.isoformat(datetime.now()),
-                }
-            )
+
+            for step, loss_val in step_loss_log.items():
+
+                # Below is added to be propagated and stored as training_metadata
+                training_loss_tracker.append(
+                    {
+                        "epoch": epoch,
+                        "step": step,
```

Comment:
This is to start logging step wise instead of per epoch. This will change a bit with Trainer, but the general format of logging would remain same.